.github/workflows/tests.yaml

```diff
@@ -0,0 +1,86 @@
+name: Test
+
+permissions:
+  contents: read
+
+on: [push, pull_request]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}
+
+jobs:
+  test:
+    name: "Tests (${{ matrix.compiler }}, C++${{ matrix.standard}}, ${{ matrix.mode }}, ${{ matrix.enables }})"
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        # only the compilers supported by setup-cpp
+        compiler: [clang++-18, gcc-13]
+        standard: [20, 23]
+        mode: [dev, debug, release]
+        include:
+          # only build this combination with dpdk enabled, so we don't double
+          # the size of the test matrix, and can at least test the build with
+          # dpdk enabled.
+          - compiler: clang++-18
+            standard: 23
+            mode: release
+            cooks: --cook dpdk
+            enables: --enable-dpdk
+          # only build this combination with C++20 moduels enabled, so we don't double
+          # the size of the test matrix, and can at least test the build with
+          # C++20 modules enabled.
+          - compiler: clang++-18
+            standard: 23
+            mode: debug
+            enables: --enable-cxx-modules
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: "${{ contains(matrix.cooks, 'dpdk') }}"
+
+      - name: Install build dependencies
+        run: |
+          sudo ./install-dependencies.sh
```

Comment:
Does this happen every single time from scratch?
Does it happen 12 times, or just once for all the builds?

I guess the question I have is how time do all these builds take, and who actually pays for them?

Reply:
> Does this happen every single time from scratch? 

yes.

> Does it happen 12 times, or just once for all the builds?

12 times.

> 
> I guess the question I have is how time do all these builds take, and who actually pays for them?

i am not sure what the "time" implies. in case of my workflow, 

- if i am not pending on the build, i don't pay for it, neither does my employer
- if i am not working on this project at my daytime, and i am staring at it, i pay for it.
- if i am not working on this project at my daytime, and i am not waiting for it, i don't pay for it.

Reply:
I didn't ask who is paying for this development or your time waiting for a build :-) I meant who is paying for the the 34 minutes (!) of CPU work and 12 times of installing the same packages over and over and over. You answered it in the last line - I guess Microsoft pays for it ;-)